example/src/main/scala/fommil/dda/main.scala

```diff
@@ -49,45 +56,94 @@ object Main extends SafeApp {
     } yield ()
   }.run.swallowError
 
-  // runs the app, requires that refresh tokens are provided
-  def agents(bearer: BearerToken): Task[Unit] = {
-    type HT[f[_], a] = EitherT[f, JsonClient.Error, a]
-    type GT[f[_], a] = StateT[f, BearerToken, a]
-    type FT[f[_], a] = StateT[f, WorldView, a]
-
-    type H[a] = HT[Task, a]
-    type G[a] = GT[H, a]
-    type F[a] = FT[G, a]
+  type JErr = JsonClient.Error :+: Throwable :+: CNil
+  type JIO[A] = IO[JErr, A]
 
-    val T: LocalClock[G] = {
-      import LocalClock.liftM
-      liftM(liftM(new LocalClockTask))
+  def unliftMonadError[F[_], E <: Cpr: Inj[?, E1]: Sel[?, E1], E1](implicit me: MonadError[F, E]): MonadError[F, E1] = new MonadError[F, E1] {
+    override def raiseError[A](e: E1): F[A] = me.raiseError[A](Cpr[E].apply(e))
+    override def handleError[A](fa: F[A])(f: E1 => F[A]): F[A] = me.handleError(fa) {
+      c => c.select[E1].fold(me.raiseError[A](c))(f)
     }
+    override def point[A](a: => A): F[A] = me.point(a)
+    override def bind[A, B](fa: F[A])(f: A => F[B]): F[B] = me.bind(fa)(f)
+  }
 
-    val S: Sleep[F] = {
-      import Sleep.liftM
-      liftM(liftM(liftM(new SleepTask)))
+  def unliftMonadIO[E <: Cpr: Inj[?, E1], E1](implicit mio: MonadIO[IO[E, ?], E]): MonadIO[IO[E, ?], E1] = new MonadIO[IO[E, ?], E1] {
+    override def liftIO[A](io: IO[E1, A])(implicit M: Monad[IO[E, ?]]): IO[E, A] = io.liftErr
+  }
+
+  def refState[E, A](a: A): Task[MonadState[IO[E, ?], A]] = IORef(a).map { ref =>
```

Comment:
these anonymous implementations of `Monad*` seem to be the bulk of the new boilerplate.